lib/auth/resetpasswordtoken.go

```diff
@@ -129,6 +129,41 @@ func (s *AuthServer) CreateResetPasswordToken(ctx context.Context, req CreateRes
 	return s.GetResetPasswordToken(ctx, token.GetName())
 }
 
+// proxyDomainGetter is a reduced subset of the Auth API for formatAccountName.
+type proxyDomainGetter interface {
+	GetProxies() ([]services.Server, error)
+	GetDomainName() (string, error)
+}
+
+// formatAccountName builds the account name to display in OTP applications.
+// Format for accountName is user@address. Use is passed in, this function
+// tries to find the best available address.
+func formatAccountName(s proxyDomainGetter, username string, authHostname string) (string, error) {
+	// Get a list of proxies. If any of the proxies has a public address set,
+	// use that. If none of the proxies have a public address set, use the
+	// hostname of the first proxy found.
+	proxies, err := s.GetProxies()
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	proxyHost, _, err := services.GuessProxyHostAndVersion(proxies)
+	if err != nil && !trace.IsNotFound(err) {
+		return "", trace.Wrap(err)
+	}
+
+	// If no proxies were found, try and set address to the name of the cluster.
+	// If even the cluster name is not found (an unlikely) event, fallback to
+	// hostname of the auth server.
+	if trace.IsNotFound(err) {
```

Comment:
nit: it would be more understandable to check this before calling `GuessProxyHostAndVersion`.
I was confused at first why that func would return `NotFound` but `GetProxies` wouldn't.